packages/reactivity/src/reactive.ts

```diff
@@ -252,24 +252,16 @@ function createReactiveObject(
     }
     return target
   }
-  // target is already a Proxy, return it.
-  // exception: calling readonly() on a reactive object
-  if (
-    target[ReactiveFlags.RAW] &&
-    !(isReadonly && target[ReactiveFlags.IS_REACTIVE])
-  ) {
+  const targetType = getTargetType(target)
```

Comment:
https://deploy-preview-10419--vue-sfc-playground.netlify.app/#eNp9UstOwzAQ/JWVLwWpJGrhFKWVAPUAB4qAoy95bFOXxLZsJ1RE+XfWrtIWhHrzzsyOZqzt2b3WUdciS1hqCyO0A4uu1UsuRaOVcdCDwaxwokMYYGNUAxPST7jkslDSOmhsBQvos6TPE5jMJsPAZRyDbh0ICUVWbPFMOiftaHhF8zWX63yHhYs2BvE7YHMCyaFUIJUD2i8+f/uo3KLpsPzj5ffS+FCDCtDgsNF15pAmAEi3s2Xfn7aHIY0J8lwqpA/c3TSqxHrB2SiKsijnDGJSpfGZHZsyZynNRlTRzipJH9h7I84K1WhRo1lrJygtZwkExnNZXauv54A50+J0xEPHf/Cd3XuMs1eDIQ9nR85lpkJ3oFfvL7in95GkGm1N6gvkG1pVtz7jQfbQypJin+lC2qdwBkJWH3a1dyjtWMoH9coh6Dmjq3i8UP0U9za6C3tcDmz4Ab7Q11g=

In the playground, using `Object.freeze` to handle proxy objects will result in an error. There should be very few people who would freeze a proxy object.

Reply:
Yeah, `Object.freeze()` is rarely used. I believe that instead of checking the cache first, we should ensure that the object is not `INVALID` before considering the subsequent actions.

Additionally, you are using `Object.freeze(msg2)` instead of `Object.freeze(msg)`.

Although this example is peculiar, in case there is a large amount of code within a single `SFC`, maintained by multiple people, if someone misuses `Object.freeze()` and another person proxies the original object, an error would be thrown in the now Vue version. However, if the object is `INVALID`, just return the original object directly, in your case, the error won't occur.